pkg/datastore/datastore.go

```diff
@@ -28,15 +28,25 @@ func New() *memoryDatastore {
 
 // Reader is the interface that wraps the Read method.
 type Reader interface {
-	// Read takes a package identifier and returns metadata defined in the opsrc
+	// Read takes an OperatorSource name and package identifier and returns metadata defined in the opsrc
 	// spec to associate that package to a specific opsrc in the marketplace.
 	// The OpsrcRef returned can be used to determine how to download manifests
 	// for a specific operator package.
-	Read(packageID string) (opsrcMeta *OpsrcRef, err error)
+	Read(source, packageID string) (opsrcMeta *OpsrcRef, err error)
 
-	// ReadVersion takes a package identifer and returns version metadata
+	// ReadByPackageName takes a package identifier and returns metadata defined in the opsrc
+	// spec to associate that package to a specific opsrc in the marketplace.
+	// The OpsrcRef returned can be used to determine how to download manifests
+	// for a specific operator package.
+	ReadByPackageName(packageID string) (opsrcMeta *OpsrcRef, err error)
+
+	// ReadRepositoryVersion takes an OperatorSource name and package identifer and returns version metadata
```

Comment:
It just returns the version. There is no other meta data here.

Reply:
Good catch - I will update this description and the original description of `Read` as well (from which this was copied from).